### ISSUE TITLE
Slow consumer causes messages delay in other clients

### ISSUE THREAD

**Issue by cpiotr** [nats-io/nats-server#2631]
Hi, 
in the scenario described below I observe a single slow consumer causing increased message delays in _other_ clients. 
Is this a known problem or can this be misconfiguration on my side? 

### Clients
* 4 clients (ids 1-4) publishing requests at 50 messages / second each
* 1 client (id=5) receiving all requests and publishing confirmations
* 4 clients (ids 6-9) receiving all the confirmations 
* 1 slow client (id=10) connecting through slow TCP proxy (200ms delay on incoming packets) 
* other clients (not more than 3) receiving all the confirmations 

### Server 
* `write_deadline: 1s`
* Throughput (number of messages): in=300 out=800
* Process memory: 23MB - 27MB
* Excerpt from logs:
```
> grep "10/15" nats-server.log  | grep -v DBG
[1] 2021/10/15 06:57:43.213420 [INF] 10.100.100.100:55895 - cid:128 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 1414 total bytes.
[1] 2021/10/15 07:00:01.283770 [INF] 10.100.100.100:55910 - cid:130 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 1422 total bytes.
[1] 2021/10/15 07:02:10.342284 [INF] 10.100.100.100:55951 - cid:132 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 711 total bytes.
[1] 2021/10/15 07:03:53.222685 [INF] 10.100.100.100:56004 - cid:133 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 709 total bytes.
[1] 2021/10/15 07:05:37.154264 [INF] 10.100.100.100:56337 - cid:134 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 1418 total bytes.
[1] 2021/10/15 07:07:19.867407 [INF] 10.100.100.100:56387 - cid:135 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 709 total bytes.
[1] 2021/10/15 07:08:57.542050 [INF] 10.100.100.100:56393 - cid:136 - "v2.12.0:id=10" - Slow Consumer Detected: WriteDeadline of 1s exceeded with 1 chunks of 709 total bytes.
```

### Observed result:
* client (id=5) timestamps outgoing messages
* clients (ids 6-9) report delay between that timestamp and the current time
* around the time the NATS Server reports the slow consumer, the delays reported by the clients are slightly exceeding 1000ms (typically 1001 or 1002 ms).

**Comment by kozlovic** [comment]
I am going to close this issue since it is likely that it will be addressed by PR #2684 which has just been merged. If you still reproduce issues with the new code, we will reopen this.

**Comment by cpiotr** [comment]
I've verified that changes mentioned by @kozlovic (PR #2684) indeed resolved the problem described in this Issue. 
Thanks!
